src/RA_Achievement.h

```diff
@@ -24,10 +22,33 @@ enum Achievement_DirtyFlags
     Dirty__All = (unsigned int)(-1)
 };
 
+#ifdef RA_UTEST
+namespace ra {
+namespace data {
+namespace tests {
+
+class RA_Achievement_Tests;
+
+} // namespace ra
+} // namespace data
+} // namespace tests
+#endif
+
 class Achievement
 {
+protected:
+#ifdef RA_UTEST
+    friend class ra::data::tests::RA_Achievement_Tests;
+#endif
+    enum class Type
+    {
+        Core,
+        Unofficial,
+        Local
+    };
```

Comment:
Had to make it a type member of achievement to satisfy other requirements. Made `protected `just in case we want to subclass it later and made only the classes that need it friends. All other classes use the `AchievementSet::Type` alias.